File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-ird-5-4/InflationRateCalculation.cpp
// InflationRateCalculation.cpp 
#include "InflationRateCalculation.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

InflationRateCalculation::InflationRateCalculation(TiXmlNode* xmlNode)
: FloatingRateCalculation(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //inflationLagNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* inflationLagNode = xmlNode->FirstChildElement("inflationLag");

   if(inflationLagNode){inflationLagIsNull_ = false;}
   else{inflationLagIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- inflationLagNode , address : " << inflationLagNode << std::endl;
   #endif
   if(inflationLagNode)
   {
      if(inflationLagNode->Attribute("href") || inflationLagNode->Attribute("id"))
      {
          if(inflationLagNode->Attribute("id"))
          {
             inflationLagIDRef_ = inflationLagNode->Attribute("id");
             inflationLag_ = boost::shared_ptr<Offset>(new Offset(inflationLagNode));
             inflationLag_->setID(inflationLagIDRef_);
             IDManager::instance().setID(inflationLagIDRef_,inflationLag_);
          }
          else if(inflationLagNode->Attribute("href")) { inflationLagIDRef_ = inflationLagNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { inflationLag_ = boost::shared_ptr<Offset>(new Offset(inflationLagNode));}
   }

   //indexSourceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* indexSourceNode = xmlNode->FirstChildElement("indexSource");

   if(indexSourceNode){indexSourceIsNull_ = false;}
   else{indexSourceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- indexSourceNode , address : " << indexSourceNode << std::endl;
   #endif
   if(indexSourceNode)
   {
      if(indexSourceNode->Attribute("href") || indexSourceNode->Attribute("id"))
      {
          if(indexSourceNode->Attribute("id"))
          {
             indexSourceIDRef_ = indexSourceNode->Attribute("id");
             indexSource_ = boost::shared_ptr<RateSourcePage>(new RateSourcePage(indexSourceNode));
             indexSource_->setID(indexSourceIDRef_);
             IDManager::instance().setID(indexSourceIDRef_,indexSource_);
          }
          else if(indexSourceNode->Attribute("href")) { indexSourceIDRef_ = indexSourceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { indexSource_ = boost::shared_ptr<RateSourcePage>(new RateSourcePage(indexSourceNode));}
   }

   //mainPublicationNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* mainPublicationNode = xmlNode->FirstChildElement("mainPublication");

   if(mainPublicationNode){mainPublicationIsNull_ = false;}
   else{mainPublicationIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- mainPublicationNode , address : " << mainPublicationNode << std::endl;
   #endif
   if(mainPublicationNode)
   {
      if(mainPublicationNode->Attribute("href") || mainPublicationNode->Attribute("id"))
      {
          if(mainPublicationNode->Attribute("id"))
          {
             mainPublicationIDRef_ = mainPublicationNode->Attribute("id");
             mainPublication_ = boost::shared_ptr<MainPublication>(new MainPublication(mainPublicationNode));
             mainPublication_->setID(mainPublicationIDRef_);
             IDManager::instance().setID(mainPublicationIDRef_,mainPublication_);
          }
          else if(mainPublicationNode->Attribute("href")) { mainPublicationIDRef_ = mainPublicationNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { mainPublication_ = boost::shared_ptr<MainPublication>(new MainPublication(mainPublicationNode));}
   }

   //interpolationMethodNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* interpolationMethodNode = xmlNode->FirstChildElement("interpolationMethod");

   if(interpolationMethodNode){interpolationMethodIsNull_ = false;}
   else{interpolationMethodIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- interpolationMethodNode , address : " << interpolationMethodNode << std::endl;
   #endif
   if(interpolationMethodNode)
   {
      if(interpolationMethodNode->Attribute("href") || interpolationMethodNode->Attribute("id"))
      {
          if(interpolationMethodNode->Attribute("id"))
          {
             interpolationMethodIDRef_ = interpolationMethodNode->Attribute("id");
             interpolationMethod_ = boost::shared_ptr<InterpolationMethod>(new InterpolationMethod(interpolationMethodNode));
             interpolationMethod_->setID(interpolationMethodIDRef_);
             IDManager::instance().setID(interpolationMethodIDRef_,interpolationMethod_);
          }
          else if(interpolationMethodNode->Attribute("href")) { interpolationMethodIDRef_ = interpolationMethodNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { interpolationMethod_ = boost::shared_ptr<InterpolationMethod>(new InterpolationMethod(interpolationMethodNode));}
   }

   //initialIndexLevelNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* initialIndexLevelNode = xmlNode->FirstChildElement("initialIndexLevel");

   if(initialIndexLevelNode){initialIndexLevelIsNull_ = false;}
   else{initialIndexLevelIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- initialIndexLevelNode , address : " << initialIndexLevelNode << std::endl;
   #endif
   if(initialIndexLevelNode)
   {
      if(initialIndexLevelNode->Attribute("href") || initialIndexLevelNode->Attribute("id"))
      {
          if(initialIndexLevelNode->Attribute("id"))
          {
             initialIndexLevelIDRef_ = initialIndexLevelNode->Attribute("id");
             initialIndexLevel_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(initialIndexLevelNode));
             initialIndexLevel_->setID(initialIndexLevelIDRef_);
             IDManager::instance().setID(initialIndexLevelIDRef_,initialIndexLevel_);
          }
          else if(initialIndexLevelNode->Attribute("href")) { initialIndexLevelIDRef_ = initialIndexLevelNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { initialIndexLevel_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(initialIndexLevelNode));}
   }

   //fallbackBondApplicableNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* fallbackBondApplicableNode = xmlNode->FirstChildElement("fallbackBondApplicable");

   if(fallbackBondApplicableNode){fallbackBondApplicableIsNull_ = false;}
   else{fallbackBondApplicableIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- fallbackBondApplicableNode , address : " << fallbackBondApplicableNode << std::endl;
   #endif
   if(fallbackBondApplicableNode)
   {
      if(fallbackBondApplicableNode->Attribute("href") || fallbackBondApplicableNode->Attribute("id"))
      {
          if(fallbackBondApplicableNode->Attribute("id"))
          {
             fallbackBondApplicableIDRef_ = fallbackBondApplicableNode->Attribute("id");
             fallbackBondApplicable_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(fallbackBondApplicableNode));
             fallbackBondApplicable_->setID(fallbackBondApplicableIDRef_);
             IDManager::instance().setID(fallbackBondApplicableIDRef_,fallbackBondApplicable_);
          }
          else if(fallbackBondApplicableNode->Attribute("href")) { fallbackBondApplicableIDRef_ = fallbackBondApplicableNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { fallbackBondApplicable_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(fallbackBondApplicableNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<Offset> InflationRateCalculation::getInflationLag()
{
   if(!this->inflationLagIsNull_){
        if(inflationLagIDRef_ != ""){
             return boost::shared_static_cast<Offset>(IDManager::instance().getID(inflationLagIDRef_));
        }else{
             return this->inflationLag_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Offset>();
   }
}
boost::shared_ptr<RateSourcePage> InflationRateCalculation::getIndexSource()
{
   if(!this->indexSourceIsNull_){
        if(indexSourceIDRef_ != ""){
             return boost::shared_static_cast<RateSourcePage>(IDManager::instance().getID(indexSourceIDRef_));
        }else{
             return this->indexSource_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<RateSourcePage>();
   }
}
boost::shared_ptr<MainPublication> InflationRateCalculation::getMainPublication()
{
   if(!this->mainPublicationIsNull_){
        if(mainPublicationIDRef_ != ""){
             return boost::shared_static_cast<MainPublication>(IDManager::instance().getID(mainPublicationIDRef_));
        }else{
             return this->mainPublication_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<MainPublication>();
   }
}
boost::shared_ptr<InterpolationMethod> InflationRateCalculation::getInterpolationMethod()
{
   if(!this->interpolationMethodIsNull_){
        if(interpolationMethodIDRef_ != ""){
             return boost::shared_static_cast<InterpolationMethod>(IDManager::instance().getID(interpolationMethodIDRef_));
        }else{
             return this->interpolationMethod_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<InterpolationMethod>();
   }
}
boost::shared_ptr<XsdTypeDecimal> InflationRateCalculation::getInitialIndexLevel()
{
   if(!this->initialIndexLevelIsNull_){
        if(initialIndexLevelIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDecimal>(IDManager::instance().getID(initialIndexLevelIDRef_));
        }else{
             return this->initialIndexLevel_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDecimal>();
   }
}
boost::shared_ptr<XsdTypeBoolean> InflationRateCalculation::getFallbackBondApplicable()
{
   if(!this->fallbackBondApplicableIsNull_){
        if(fallbackBondApplicableIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(fallbackBondApplicableIDRef_));
        }else{
             return this->fallbackBondApplicable_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-ird-5-4/InflationRateCalculation.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class InflationRateCalculation : FloatingRateCalculation
    {
        public InflationRateCalculation(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode inflationLagNode = xmlNode.SelectSingleNode("inflationLag");
            
            if (inflationLagNode != null)
            {
                if (inflationLagNode.Attributes["href"] != null || inflationLagNode.Attributes["id"] != null) 
                {
                    if (inflationLagNode.Attributes["id"] != null) 
                    {
                        inflationLagIDRef_ = inflationLagNode.Attributes["id"].Value;
                        Offset ob = new Offset(inflationLagNode);
                        IDManager.SetID(inflationLagIDRef_, ob);
                    }
                    else if (inflationLagNode.Attributes["href"] != null)
                    {
                        inflationLagIDRef_ = inflationLagNode.Attributes["href"].Value;
                    }
                    else
                    {
                        inflationLag_ = new Offset(inflationLagNode);
                    }
                }
                else
                {
                    inflationLag_ = new Offset(inflationLagNode);
                }
            }
            
        
            XmlNode indexSourceNode = xmlNode.SelectSingleNode("indexSource");
            
            if (indexSourceNode != null)
            {
                if (indexSourceNode.Attributes["href"] != null || indexSourceNode.Attributes["id"] != null) 
                {
                    if (indexSourceNode.Attributes["id"] != null) 
                    {
                        indexSourceIDRef_ = indexSourceNode.Attributes["id"].Value;
                        RateSourcePage ob = new RateSourcePage(indexSourceNode);
                        IDManager.SetID(indexSourceIDRef_, ob);
                    }
                    else if (indexSourceNode.Attributes["href"] != null)
                    {
                        indexSourceIDRef_ = indexSourceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        indexSource_ = new RateSourcePage(indexSourceNode);
                    }
                }
                else
                {
                    indexSource_ = new RateSourcePage(indexSourceNode);
                }
            }
            
        
            XmlNode mainPublicationNode = xmlNode.SelectSingleNode("mainPublication");
            
            if (mainPublicationNode != null)
            {
                if (mainPublicationNode.Attributes["href"] != null || mainPublicationNode.Attributes["id"] != null) 
                {
                    if (mainPublicationNode.Attributes["id"] != null) 
                    {
                        mainPublicationIDRef_ = mainPublicationNode.Attributes["id"].Value;
                        MainPublication ob = new MainPublication(mainPublicationNode);
                        IDManager.SetID(mainPublicationIDRef_, ob);
                    }
                    else if (mainPublicationNode.Attributes["href"] != null)
                    {
                        mainPublicationIDRef_ = mainPublicationNode.Attributes["href"].Value;
                    }
                    else
                    {
                        mainPublication_ = new MainPublication(mainPublicationNode);
                    }
                }
                else
                {
                    mainPublication_ = new MainPublication(mainPublicationNode);
                }
            }
            
        
            XmlNode interpolationMethodNode = xmlNode.SelectSingleNode("interpolationMethod");
            
            if (interpolationMethodNode != null)
            {
                if (interpolationMethodNode.Attributes["href"] != null || interpolationMethodNode.Attributes["id"] != null) 
                {
                    if (interpolationMethodNode.Attributes["id"] != null) 
                    {
                        interpolationMethodIDRef_ = interpolationMethodNode.Attributes["id"].Value;
                        InterpolationMethod ob = new InterpolationMethod(interpolationMethodNode);
                        IDManager.SetID(interpolationMethodIDRef_, ob);
                    }
                    else if (interpolationMethodNode.Attributes["href"] != null)
                    {
                        interpolationMethodIDRef_ = interpolationMethodNode.Attributes["href"].Value;
                    }
                    else
                    {
                        interpolationMethod_ = new InterpolationMethod(interpolationMethodNode);
                    }
                }
                else
                {
                    interpolationMethod_ = new InterpolationMethod(interpolationMethodNode);
                }
            }
            
        
            XmlNode initialIndexLevelNode = xmlNode.SelectSingleNode("initialIndexLevel");
            
            if (initialIndexLevelNode != null)
            {
                if (initialIndexLevelNode.Attributes["href"] != null || initialIndexLevelNode.Attributes["id"] != null) 
                {
                    if (initialIndexLevelNode.Attributes["id"] != null) 
                    {
                        initialIndexLevelIDRef_ = initialIndexLevelNode.Attributes["id"].Value;
                        XsdTypeDecimal ob = new XsdTypeDecimal(initialIndexLevelNode);
                        IDManager.SetID(initialIndexLevelIDRef_, ob);
                    }
                    else if (initialIndexLevelNode.Attributes["href"] != null)
                    {
                        initialIndexLevelIDRef_ = initialIndexLevelNode.Attributes["href"].Value;
                    }
                    else
                    {
                        initialIndexLevel_ = new XsdTypeDecimal(initialIndexLevelNode);
                    }
                }
                else
                {
                    initialIndexLevel_ = new XsdTypeDecimal(initialIndexLevelNode);
                }
            }
            
        
            XmlNode fallbackBondApplicableNode = xmlNode.SelectSingleNode("fallbackBondApplicable");
            
            if (fallbackBondApplicableNode != null)
            {
                if (fallbackBondApplicableNode.Attributes["href"] != null || fallbackBondApplicableNode.Attributes["id"] != null) 
                {
                    if (fallbackBondApplicableNode.Attributes["id"] != null) 
                    {
                        fallbackBondApplicableIDRef_ = fallbackBondApplicableNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(fallbackBondApplicableNode);
                        IDManager.SetID(fallbackBondApplicableIDRef_, ob);
                    }
                    else if (fallbackBondApplicableNode.Attributes["href"] != null)
                    {
                        fallbackBondApplicableIDRef_ = fallbackBondApplicableNode.Attributes["href"].Value;
                    }
                    else
                    {
                        fallbackBondApplicable_ = new XsdTypeBoolean(fallbackBondApplicableNode);
                    }
                }
                else
                {
                    fallbackBondApplicable_ = new XsdTypeBoolean(fallbackBondApplicableNode);
                }
            }
            
        
        }
        
    
        #region InflationLag_
        private Offset inflationLag_;
        public Offset InflationLag_
        {
            get
            {
                if (this.inflationLag_ != null)
                {
                    return this.inflationLag_; 
                }
                else if (this.inflationLagIDRef_ != null)
                {
                    inflationLag_ = IDManager.getID(inflationLagIDRef_) as Offset;
                    return this.inflationLag_; 
                }
                else
                {
                      return this.inflationLag_; 
                }
            }
            set
            {
                if (this.inflationLag_ != value)
                {
                    this.inflationLag_ = value;
                }
            }
        }
        #endregion
        
        public string inflationLagIDRef_ { get; set; }
        #region IndexSource_
        private RateSourcePage indexSource_;
        public RateSourcePage IndexSource_
        {
            get
            {
                if (this.indexSource_ != null)
                {
                    return this.indexSource_; 
                }
                else if (this.indexSourceIDRef_ != null)
                {
                    indexSource_ = IDManager.getID(indexSourceIDRef_) as RateSourcePage;
                    return this.indexSource_; 
                }
                else
                {
                      return this.indexSource_; 
                }
            }
            set
            {
                if (this.indexSource_ != value)
                {
                    this.indexSource_ = value;
                }
            }
        }
        #endregion
        
        public string indexSourceIDRef_ { get; set; }
        #region MainPublication_
        private MainPublication mainPublication_;
        public MainPublication MainPublication_
        {
            get
            {
                if (this.mainPublication_ != null)
                {
                    return this.mainPublication_; 
                }
                else if (this.mainPublicationIDRef_ != null)
                {
                    mainPublication_ = IDManager.getID(mainPublicationIDRef_) as MainPublication;
                    return this.mainPublication_; 
                }
                else
                {
                      return this.mainPublication_; 
                }
            }
            set
            {
                if (this.mainPublication_ != value)
                {
                    this.mainPublication_ = value;
                }
            }
        }
        #endregion
        
        public string mainPublicationIDRef_ { get; set; }
        #region InterpolationMethod_
        private InterpolationMethod interpolationMethod_;
        public InterpolationMethod InterpolationMethod_
        {
            get
            {
                if (this.interpolationMethod_ != null)
                {
                    return this.interpolationMethod_; 
                }
                else if (this.interpolationMethodIDRef_ != null)
                {
                    interpolationMethod_ = IDManager.getID(interpolationMethodIDRef_) as InterpolationMethod;
                    return this.interpolationMethod_; 
                }
                else
                {
                      return this.interpolationMethod_; 
                }
            }
            set
            {
                if (this.interpolationMethod_ != value)
                {
                    this.interpolationMethod_ = value;
                }
            }
        }
        #endregion
        
        public string interpolationMethodIDRef_ { get; set; }
        #region InitialIndexLevel_
        private XsdTypeDecimal initialIndexLevel_;
        public XsdTypeDecimal InitialIndexLevel_
        {
            get
            {
                if (this.initialIndexLevel_ != null)
                {
                    return this.initialIndexLevel_; 
                }
                else if (this.initialIndexLevelIDRef_ != null)
                {
                    initialIndexLevel_ = IDManager.getID(initialIndexLevelIDRef_) as XsdTypeDecimal;
                    return this.initialIndexLevel_; 
                }
                else
                {
                      return this.initialIndexLevel_; 
                }
            }
            set
            {
                if (this.initialIndexLevel_ != value)
                {
                    this.initialIndexLevel_ = value;
                }
            }
        }
        #endregion
        
        public string initialIndexLevelIDRef_ { get; set; }
        #region FallbackBondApplicable_
        private XsdTypeBoolean fallbackBondApplicable_;
        public XsdTypeBoolean FallbackBondApplicable_
        {
            get
            {
                if (this.fallbackBondApplicable_ != null)
                {
                    return this.fallbackBondApplicable_; 
                }
                else if (this.fallbackBondApplicableIDRef_ != null)
                {
                    fallbackBondApplicable_ = IDManager.getID(fallbackBondApplicableIDRef_) as XsdTypeBoolean;
                    return this.fallbackBondApplicable_; 
                }
                else
                {
                      return this.fallbackBondApplicable_; 
                }
            }
            set
            {
                if (this.fallbackBondApplicable_ != value)
                {
                    this.fallbackBondApplicable_ = value;
                }
            }
        }
        #endregion
        
        public string fallbackBondApplicableIDRef_ { get; set; }
        
    
        
    
    }
    
}

